libs/core/langchain_core/messages/ai.py

```diff
@@ -47,6 +48,87 @@ def __add__(self, other: Any) -> BaseMessageChunk:  # type: ignore
                     "Cannot concatenate AIMessageChunks with different example values."
                 )
 
+            content = merge_content(self.content, other.content)
+            additional_kwargs = merge_dicts(
+                self.additional_kwargs, other.additional_kwargs
+            )
+            response_metadata = merge_dicts(
+                self.response_metadata, other.response_metadata
+            )
+
+            if isinstance(other, AIToolCallsMessageChunk):
+                return AIToolCallsMessageChunk(
+                    example=self.example,
+                    content=content,
+                    additional_kwargs=additional_kwargs,
+                    response_metadata=response_metadata,
+                    tool_call_chunks=other.tool_call_chunks,
+                    id=self.id,
+                )
+
+            return self.__class__(
+                example=self.example,
+                content=content,
+                additional_kwargs=additional_kwargs,
+                response_metadata=response_metadata,
+                id=self.id,
+            )
+
+        return super().__add__(other)
+
+
+class ToolCall(Serializable):
+    name: str
+    args: dict
+    id: Optional[str] = None
+    index: Optional[int] = None
```

Comment:
@baskaryan how do you feel about removing `index` from ToolCall? We need it on ToolCallChunk for merging but I'm not sure it's needed on ToolCall.